### DISCUSSION TITLE
Volta 1.1.1 update breaks syslinks

### DISCUSSION THREAD

**Discussion by MatthiasPortzel** [Homebrew/discussions#4169] [Casks]
<details>
<summary> brew config and brew doctor output </summary>
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.20-167-g7a5f614
ORIGIN: https://github.com/Homebrew/brew
HEAD: 7a5f6143d3e7447a64255c548d2e51138c83acd3
Last commit: 3 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 9e5a23de71f24a17ee1b0652d3331f41fe639aa0
Core tap last commit: 8 days ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.NW0lYcm3Uf/org.xquartz:0
HOMEBREW_EDITOR: emacsclient
HOMEBREW_INSTALL_FROM_API: set
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit kabylake
Clang: 14.0.0 build 1400
Git: 2.39.1 => /usr/local/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.6.2-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
```

### Output of `brew doctor`

```shell
﻿﻿﻿﻿Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed '.pc' files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected '.pc' files:
  /usr/local/lib/pkgconfig/fuse-ext2.pc

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  six

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc
```
</details>

### Description of issue

After updating from Volta 1.1.0 to 1.1.1, Volta's syslinks in ~/.volta still pointed to the old version.

```bash
MacBookGamma:~ | matthias% node                                                                                                                  20:14
zsh: command not found: node
MacBookGamma:~ | matthias% type node                                                                                                             20:14
node not found
MacBookGamma:~ | matthias% echo $path                                                                                                            20:14
... /Users/matthias/.volta/bin ...
MacBookGamma:~ | matthias% l -l /Users/matthias/.volta/bin                                                                                       20:14
lrwxr-xr-x 44 matthias 28 Nov  2022 node -> /usr/local/Cellar/volta/1.1.0/bin/volta-shim
lrwxr-xr-x 44 matthias 28 Nov  2022 npm -> /usr/local/Cellar/volta/1.1.0/bin/volta-shim
lrwxr-xr-x 44 matthias 28 Nov  2022 npx -> /usr/local/Cellar/volta/1.1.0/bin/volta-shim
lrwxr-xr-x 44 matthias 28 Nov  2022 yarn -> /usr/local/Cellar/volta/1.1.0/bin/volta-shim
```

Deleting my ~/.volta and running `volta install node@18` re-created the .volta folder with the correct syslinks (same as above just 1.1.1 instead of 1.1.0).

My brew setup is slightly non-standard, but wanted to post here to document this behavior in case other people experience it.

### Relevant casks

volta

**Comment by gromgit** [comment]
What's the output of the following?
```
export VOLTA_INSTALL_DIR=$(brew --prefix dvc)/bin
volta install node@18
ls -l ~/.volta/bin
```

**Reply by gromgit** [reply]
Also see https://github.com/volta-cli/volta/issues/1053.

**Reply by carlocab** [reply]
That should probably be
```
export VOLTA_INSTALL_DIR=$(brew --prefix)/bin
```

**Reply by cho-m** [reply]
Not sure on impact, but could build formula with env scripts. If moving binaries to libexec, then maybe `VOLTA_INSTALL_DIR: "${VOLTA_INSTALL_DIR:-#{opt_libexec}}"`.

**Reply by gromgit** [reply]
> That should probably be
> 
> ```
> export VOLTA_INSTALL_DIR=$(brew --prefix)/bin
> ```

That'll teach me to juggle two different issues at the same time. :man_facepalming:

**Reply by gromgit** [reply]
> Not sure on impact, but could build formula with env scripts. If moving binaries to libexec, then maybe `VOLTA_INSTALL_DIR: "${VOLTA_INSTALL_DIR:-#{opt_libexec}}"`.

Opened https://github.com/Homebrew/homebrew-core/pull/122278 to fix this.

**Reply by gromgit** [reply]
@MatthiasPortzel , a fix has been committed. `brew update`, `brew upgrade volta`, then `volta-migrate` and check the links in your `~/.volta/bin` directory.

**Reply by MatthiasPortzel** [reply]
@gromgit Thanks for looking into this. After running those commands:
```sh
% ls -l ~/.volta/bin
total 0
lrwxr-xr-x  1 matthias  staff  35 Feb  4 13:30 node -> /usr/local/opt/volta/bin/volta-shim
lrwxr-xr-x  1 matthias  staff  35 Feb  4 13:30 npm -> /usr/local/opt/volta/bin/volta-shim
lrwxr-xr-x  1 matthias  staff  35 Feb  4 13:30 npx -> /usr/local/opt/volta/bin/volta-shim
lrwxr-xr-x  1 matthias  staff  35 Feb  4 13:30 pnpm -> /usr/local/opt/volta/bin/volta-shim
lrwxr-xr-x  1 matthias  staff  35 Feb  4 13:30 yarn -> /usr/local/opt/volta/bin/volta-shim
```
But!

```
% type node
node is /Users/matthias/.volta/bin/node

% node
Volta error: 'volta-shim' should not be called directly.

Please use the existing shims provided by Volta (node, yarn, etc.) to run tools.
```
So unfortunately this isn't working.

**Reply by cho-m** [reply]
Sadly, looks like `volta` doesn't like env scripts. Probably needs the name of original command (e.g. `node`) passed through. May need to consider another alternative.

If this is only an issue with `volta-shim`, there is an option of avoiding env script on that one (as long as it doesn't need `VOLTA_INSTALL_DIR` set). Otherwise, could see if `volta-migrate` postinstall could work or some other intermediary script options.

**Reply by svenjacobs** [reply]
Having the same issue as @MatthiasPortzel. Currently my Volta installed via brew is broken.

**Reply by dsully** [reply]
+1

**Reply by gromgit** [reply]
> If this is only an issue with `volta-shim`, there is an option of avoiding env script on that one (as long as it doesn't need `VOLTA_INSTALL_DIR` set).

Thanks @cho-m, that seems to be sufficient. Followup PR inbound: https://github.com/Homebrew/homebrew-core/pull/122350

**Reply by MatthiasPortzel** [reply]
Everything seems to be working fine now. Thank you!

**Reply by svenjacobs** [reply]
Yes, can also confirm Volta is working again. Thank you 👍🏼
